### PR TITLE
wait_for_success

### PR DESCRIPTION
**Pull request by hrpatel**
- Fix unbound variable
- Update comments for `wait_for_file`
- Update variable names for `wait_for_file`
- Fix `max_count` calculation for `wait_for_file`
- Implement `wait_for_success`

### CODE REVIEW

File: shtdlib.sh [hunk 1]
```diff
@@ -639,21 +639,42 @@ function resolve_srv_name {
...
+#  - third param: sleep interval (optional, default 1 sec)
+function wait_for_success {
+    local command="${1}"
+    local timeout="${2:-10}"
+    local sleep_interval="${3:-1}"
+    local max_count=$((timeout/sleep_interval))
+    local count=0
+    while ! ${command}; do
+        (( count++ ))
+        if [ ${count} -ge ${max_count} ] ; then
```
Comment (mroote) [inline shtdlib.sh]: Nitpicky but do we leave a space before the ; or remove it? I see both in the commit

**Comment by mroote** [comment]
+1 just a style comment

File: shtdlib.sh [hunk 2]
```diff
@@ -639,21 +639,42 @@ function resolve_srv_name {
...
+        fi
+    done
+}
+
+# Wait for a command to return a 0 exit status
+#  - first param: command
+#  - second param: timeout (optional, default 10 sec)
+#  - third param: sleep interval (optional, default 1 sec)
+function wait_for_success {
+    local command="${1}"
```
Comment (mroote) [inline shtdlib.sh]: What's the behaviour if the first param isn't provided? Should maybe return an error

Comment (hrpatel) [reply shtdlib.sh]: Fixed

**Comment by adamgilbert** [comment]
+1

**Pull request merged**